src/veins/base/phyLayer/BasePhyLayer.h

```diff
@@ -71,224 +63,153 @@ class Radio;
      */
     void initializeAntenna(cXMLElement* xmlConfig);
 
 protected:
     /**
-     * @brief Reads and returns the parameter with the passed name.
+     * Read and return the parameter with the passed name.
      *
-     * If the parameter couldn't be found the value of defaultValue
-     * is returned.
+     * If the parameter couldn't be found the value of defaultValue is returned.
      *
-     * @param parName         - the name of the ned-parameter
-     * @param defaultValue     - the value to be returned if the parameter couldn't be found
+     * @param parName The name of the ned-parameter
+     * @param defaultValue The value to be returned if the parameter couldn't be found
      */
     template <class T>
     T readPar(const char* parName, const T defaultValue);
 
     /**
-     * @brief OMNeT++ initialization function.
-     *
-     * Read simple parameters.
-     * Read and parse xml file for decider and analogue models
-     * configuration.
+     * Initialize members and associated objects according to parameterization.
      */
     void initialize(int stage) override;
 
     /**
-     * @brief OMNeT++ handle message function.
+     * Process received messages and pass to specific handlers.
      *
-     * Classify and forward message to subroutines.
-     * - AirFrames from channel
-     * - self scheduled AirFrames
-     * - MacPackets from MAC layer
-     * - ControllMesasges from MAC layer
-     * - self messages like TX_OVER and RADIO_SWITCHED
+     * @see handleSelfMessage
+     * @see handleUpperMessage
+     * @see handleUpperControlMessage
+     * @see handleAirFrame
      */
     void handleMessage(cMessage* msg) override;
 
     /**
-     * @brief Initializes and returns the radio class to use.
-     *
-     * Can be overridden by sub-classing phy layers to use their
-     * own Radio implementations.
+     * Create and return the radio class to use.
      */
-    virtual Radio* initializeRadio();
+    virtual std::unique_ptr<Radio> initializeRadio();
 
     /**
-     * @brief Creates and returns an instance of the AnalogueModel with the
+     * Create and return an instance of the AnalogueModel with the
      * specified name.
      *
-     * The returned AnalogueModel has to be
-     * generated with the "new" command. The BasePhyLayer
-     * keeps the ownership of the returned AnalogueModel.
-     *
-     * This method is used by the BasePhyLayer during
-     * initialisation to load the AnalogueModels which
-     * has been specified in the ned file.
-     *
-     * This method has to be overridden if you want to be
-     * able to load your own AnalogueModels.
+     * This method is used by the BasePhyLayer during initialisation to load the AnalogueModels which has been specified in the NED file.
      */
-    virtual AnalogueModel* getAnalogueModelFromName(std::string name, ParameterMap& params);
+    virtual std::unique_ptr<AnalogueModel> getAnalogueModelFromName(std::string name, ParameterMap& params)
```

Comment:
Unfortunately this would require us to make the `BasePhyLayer` a OMNeT++ module interface, which prevents us from providing default values for module parameters.